WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/UsernameChangerFullScreenDialogFragment.java

```diff
@@ -0,0 +1,322 @@
+            controller.dismiss();
+        }
+
+        return true;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean(KEY_SHOULD_WATCH_TEXT, false);
+        outState.putString(KEY_USERNAME_SELECTED, mUsernameSelected);
+        outState.putInt(KEY_USERNAME_SELECTED_INDEX, mUsernameSelectedIndex);
+        outState.putStringArrayList(KEY_USERNAME_SUGGESTIONS, new ArrayList<>(mUsernamesAdapter.mItems));
+    }
+
+    @Override
+    public void onUsernameSelected(String username) {
+        mHeaderView.setText(getHeaderText(username, mDisplayName));
+        mUsernameSelected = username;
+        mUsernameSelectedIndex = mUsernamesAdapter.getSelectedItem();
+    }
+
+    protected Spanned getHeaderText(String username, String display) {
+        return Html.fromHtml(
+                String.format(
+                        getString(R.string.username_changer_header),
+                        "<b>",
+                        username,
+                        "</b>",
+                        "<b>",
+                        display,
+                        "</b>"
+                )
+        );
+    }
+
+    protected String getUsernameOrSelected() {
+        return TextUtils.isEmpty(mUsernameSelected) ? mUsername : mUsernameSelected;
+    }
+
+    protected void getUsernameSuggestions(String usernameQuery) {
+        showProgress(true);
+
+        FetchUsernameSuggestionsPayload payload = new FetchUsernameSuggestionsPayload(usernameQuery);
+        mDispatcher.dispatch(AccountActionBuilder.newFetchUsernameSuggestionsAction(payload));
+    }
+
+    private String getUsernameQueryFromDisplayName() {
+        return mDisplayName.replace(" ", "").toLowerCase();
+    }
+
+    protected boolean hasUsernameChanged() {
+        return !TextUtils.equals(mUsername, mUsernameSelected);
+    }
+
+    private void populateUsernameSuggestions(List<String> suggestions) {
+        String usernameOrSelected = getUsernameOrSelected();
+
+        List<String> suggestionList = new ArrayList<>();
+        suggestionList.add(usernameOrSelected);
+
+        for (String suggestion : suggestions) {
+            if (!TextUtils.equals(suggestion, usernameOrSelected)) {
+                suggestionList.add(suggestion);
+            }
+        }
+
+        mUsernamesAdapter = new UsernameChangerRecyclerViewAdapter(getActivity(), suggestionList);
+        mUsernamesAdapter.setOnUsernameSelectedListener(UsernameChangerFullScreenDialogFragment.this);
+        mUsernamesAdapter.setSelectedItem(0);
+        mUsernameSelectedIndex = 0;
+        mUsernameSuggestions.setAdapter(mUsernamesAdapter);
+    }
+
+    private void setUsernameSuggestions(List<String> suggestions) {
+        mUsernamesAdapter = new UsernameChangerRecyclerViewAdapter(getActivity(), suggestions);
+        mUsernamesAdapter.setOnUsernameSelectedListener(UsernameChangerFullScreenDialogFragment.this);
```

Comment:
Qualifying the `this` with `UsernameChangerFullScreenDialogFragment` is probably redundant here, right?

Reply:
It's a little verbose, but it doesn't hurt anything and it makes things absolutely clear.